src/main/scala/org/apache/sysml/compiler/lang/source/DML.scala

```diff
@@ -0,0 +1,473 @@
+                val module = tgt(env)
+
+                // update on matrix objects (left indexing): A[r, c] = v === A.update(r, c, v)
+                val rows = args(0) // rows
+                val cols = args(1) // columns
+                val value = args(2) // value to update with
+
+                (rows, cols) match {
+                  case (":::", c) if c.contains(":")  => s"$module[,$c] = $value"
+                  case (":::", c)                     => s"$module[,$c + 1] = $value"
+                  case (r, ":::") if r.contains(":")  => s"$module[$r,] = $value"
+                  case (r, ":::")                     => s"$module[$r + 1,] = $value"
+                  case (r, c) if  r.contains(":") &&
+                    c.contains(":")     => s"$module[$r,$c] = $value"
+                  case (r, c) if r.contains(":")      => s"$module[$r, $c + 1] = $value"
+                  case (r, c) if c.contains(":")      => s"$module[$r + 1, $c] = $value"
+                  case (r, c)                         => s"$module[$r + 1,$c + 1] = $value"
+                }
+              }
+
+              // matches methods with multiple arguments (e.g. zeros(3, 3), write)
+              case (Some(tgt), (x :: xs) :: Nil) => {
+
+                val module = tgt(env)
+                val argString = args.mkString(" ")
+
+                if (module == "Matrix") {
+                  printConstructor(method, argss, env, false)
+                }
+
+                else if (module == "Vector") {
+                  printConstructor(method, argss, env, true)
+                }
+
+                else if (module == "package") {
+                  // builtin
+                  printBuiltin(tgt, method, argss, env)
+                }
+
+                else {
+                  "case (Some(tgt), (x :: xs) :: Nil)"
+                }
+              }
+
+              // matches functions without arguments (.t (transpose))
+              case (Some(tgt), Nil) => {
+
+                method.name.decodedName.toTermName match {
+                  case u.TermName(tn) if matrixFuncs.contains(tn) => s"$tn(${tgt(env)})"
+                  case u.TermName(tn) if tn == "toDouble" => tgt(env) // this is a scala implicit conversion from Int to Double
+                  case _ => method.name.decodedName.toString
+                }
+              }
+
+              case (Some(tgt), _) => abort(s"Matching error, please report the following: case case (Some(tgt), _): target ${tgt(env)}, method: $method")
+
+              // matches functions that are not defined in a module or class (udfs)
+              case (None, _) => {
+                val name = method.name.decodedName
+                val argString = args.mkString(", ")
+                s"$name($argString)"
+              }
+
+              case _ =>
+                abort(s"Unsupported function call! Calling the method $method is not supported!")
```

Comment:
Is it possible to add some location information about the original line of Scala code to this error?